packages/bsky/src/api/app/bsky/graph/muteActor.ts

```diff
@@ -18,10 +20,26 @@ export default function (server: Server, ctx: AppContext) {
         throw new InvalidRequestError('Cannot mute oneself')
       }
 
-      await ctx.services.graph(db).muteActor({
-        subjectDid,
-        mutedByDid: requester,
-      })
+      if (!ctx.cfg.bsyncOnlyMutes) {
+        await ctx.services.graph(db).muteActor({
+          subjectDid,
+          mutedByDid: requester,
+        })
+      } else {
+        assert(ctx.bsyncClient)
+      }
+
+      if (ctx.bsyncClient) {
+        try {
+          await ctx.bsyncClient.addMuteOperation({
+            type: MuteOperation_Type.ADD,
+            actorDid: requester,
+            subject: subjectDid,
+          })
+        } catch (err) {
+          req.log.warn(err, 'failed to sync mute op to bsync')
```

Comment:
i think we wanna throw here if `bsyncOnlyMutes` is true because then the operation failed

if it's false then it was partially applied but presumably the local db is considered the canonical state?

Reply:
Yep 💯 thanks for nabbing that.